bitnami/external-dns/templates/deployment.yaml

```diff
@@ -0,0 +1,203 @@
+        - --aws-zone-type={{ .Values.aws.zoneType }}
+        {{- end }}
+        {{- end }}
+
+        {{- if eq .Values.provider "azure" }}
+        {{- if .Values.azure.resoureGroup }}
+        --azure-resource-group={{ .Values.azure.resoureGroup }}
+        {{- end }}
+        {{- end }}
+
+        {{- if eq .Values.provider "google" }}
+        - --google-project={{ .Values.google.project }}
+        {{- end }}
+
+        env:
+        {{- if .Values.image.debug }}
+        - name: BASH_DEBUG
+          value: "1"
+        - name: NAMI_DEBUG
+          value: "1"
+        - name: NAMI_LOG_LEVEL
+          value: "trace8"
+        {{- end }}
+        {{- $root := . -}}
+        {{- range $key, $value := .Values.extraEnv }}
+        - name: {{ $key }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "external-dns.fullname" $root }}
+              key: {{ $key }}
+        {{- end }}
+
+        {{- if eq .Values.provider "aws" }}
+        - name: AWS_DEFAULT_REGION
+          value: {{ .Values.aws.region }}
+        {{- end }}
+
+        {{- if eq .Values.provider "cloudflare" }}
+        - name: CF_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "external-dns.fullname" . }}
+              key: cloudflare_api_key
+        - name: CF_API_EMAIL
+          value: {{ .Values.cloudflare.email | quote }}
+        {{- end }}
+
+        {{- if eq .Values.provider "google" }}
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /etc/secrets/service-account/credentials.json
+        {{- end }}
+
+        ports:
+        - containerPort: {{ .Values.service.port }}
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: {{ .Values.service.port }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+        {{- end }}
+        {{- if .Values.readinessProbe.enabled }}
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: {{ .Values.service.port }}
+          initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.readinessProbe.successThreshold }}
+          failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+          {{- end }}
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        volumeMounts:
+        {{- if eq .Values.provider "aws" }}
+        - name: aws-credentials
+          mountPath: /root/.aws
```

Comment:
I imagine this is a root container, right?

Reply:
It's not a root container... We could use env. variable to point to the path where we mount the AWS credentials (see https://docs.aws.amazon.com/cli/latest/userguide/cli-environment.html). What do you think?

Reply:
In other non root solutions we set `HOME=/`

Reply:
But it is set in the container

Reply:
yes, probably the image will try to look the aws credentials at `/.aws`. I think I would make explicit in the chart where to look for the credentials using env vars as Juan suggests